internal/tracinginterceptor/interceptor.go

```diff
@@ -39,54 +45,189 @@ var (
 
 // Params defines the parameters for creating the Interceptor
 type Params struct {
-	// Tracer is used to propagate context and to generate spans
-	Tracer opentracing.Tracer
-	// Transport is the name of the transport, it decides the propagation format and propagation carrier
+	Tracer    opentracing.Tracer
 	Transport string
+	Logger    *zap.Logger
 }
 
 // Interceptor is the tracing interceptor for all RPC types.
-// It handles both tracing observability and context propagation using OpenTracing APIs.
 type Interceptor struct {
+	tracer            opentracing.Tracer
+	transport         string
+	propagationFormat opentracing.BuiltinFormat
+	log               *zap.Logger
+}
+
+// PropagationCarrier is an interface to combine both reader and writer interface
+type PropagationCarrier interface {
+	opentracing.TextMapReader
+	opentracing.TextMapWriter
 }
 
 // New constructs a tracing interceptor with the provided parameter.
 func New(p Params) *Interceptor {
-	return &Interceptor{}
+	if p.Logger == nil {
+		p.Logger = zap.NewNop()
+	}
+	m := &Interceptor{
+		tracer:            p.Tracer,
+		transport:         p.Transport,
+		propagationFormat: getPropagationFormat(p.Transport),
+		log:               p.Logger,
+	}
+	if m.tracer == nil {
+		m.tracer = opentracing.GlobalTracer()
+	}
```

Comment:
1. Let's change "m" name to "i", as it "interceptor now"
2. Let's put `if i.logger == nil` here too. It looks weird now: tracer is checked for interceptor, when logger is for params.